src/test/java/team/serenity/logic/commands/CommandTestUtil.java

```diff
@@ -0,0 +1,104 @@
+    public static final String VALID_GRP_GROUP_A = "G04";
+    public static final String VALID_GRP_GROUP_B = "G05";
+    public static final Path VALID_PATH_GROUP_A = Paths.get("LUMINUS_GROUP_A.csv");
+    public static final Path VALID_PATH_GROUP_B = Paths.get("LUMINUS_GROUP_B.csv");
+    public static final String VALID_QN_A = "What is the deadline for the report?";
+    public static final String VALID_QN_B = "When is the consultation held?";
+
+    public static final String GRP_DESC_GROUP_A = " " + PREFIX_GRP + VALID_GRP_GROUP_A;
+    public static final String GRP_DESC_GROUP_B = " " + PREFIX_GRP + VALID_GRP_GROUP_B;
+    public static final String PATH_DESC_GROUP_A = " " + PREFIX_PATH + VALID_PATH_GROUP_A;
+    public static final String PATH_DESC_GROUP_B = " " + PREFIX_PATH + VALID_PATH_GROUP_B;
+    public static final String QN_DESC_GROUP_A = " " + PREFIX_QN + VALID_QN_A;
+    public static final String QN_DESC_GROUP_B = " " + PREFIX_QN + VALID_QN_B;
+
+    // Invalid descriptions for Serenity
+    public static final String INVALID_QN_DESC = " " + PREFIX_QN; // empty string not allowed in questions
+
+    public static final String PREAMBLE_WHITESPACE = "\t  \r  \n";
+    public static final String PREAMBLE_NON_EMPTY = "NonEmptyPreamble";
+
+    /**
+     * Executes the given {@code command}, confirms that <br> - the returned {@link CommandResult} matches {@code
+     * expectedCommandResult} <br> - the {@code actualModel} matches {@code expectedModel}
+     */
+    public static void assertCommandSuccess(Command command, Model actualModel, CommandResult expectedCommandResult,
+                                            Model expectedModel) {
+        try {
+            CommandResult result = command.execute(actualModel);
+            assertEquals(expectedCommandResult, result);
+            assertEquals(expectedModel, actualModel);
+        } catch (CommandException ce) {
+            throw new AssertionError("Execution of command should not fail.", ce);
+        }
+    }
+
+    /**
+     * Convenience wrapper to {@link #assertCommandSuccess(Command, Model, CommandResult, Model)} that takes a string
+     * {@code expectedMessage}.
+     */
+    public static void assertCommandSuccess(Command command, Model actualModel, String expectedMessage,
+        Model expectedModel) {
+        CommandResult expectedCommandResult = new CommandResult(expectedMessage);
+        assertCommandSuccess(command, actualModel, expectedCommandResult, expectedModel);
+    }
+
+    /**
+     * Executes the given {@code command}, confirms that <br> - a {@code CommandException} is thrown <br> - the
+     * CommandException message matches {@code expectedMessage} <br> - the address book, filtered person list and
+     * selected person in {@code actualModel} remain unchanged
+     */
+    public static void assertCommandFailure(Command command, Model actualModel, String expectedMessage) {
+        // TODO
+        /*
+        FOR REFERENCE (AB3)
```

Comment:
Are these comments still needed?

Reply:
Keeping them so that we will not forget to implement them in future.